File: Project Files/CarPark/Fish/Fish_Python_Start/aquarium.py
from owner import Owner


class Aquarium:
    def __init__(self):
        self.all_my_owners = []

    def add_owner(self, new_id, new_first_name, new_last_name, new_birth_date):
        new_owner = Owner(new_id, new_first_name, new_last_name, new_birth_date, self)
        self.all_my_owners.append(new_owner)

    def find_owner(self, target_owner_id):
        return next((owner for owner in self.all_my_owners if owner.id == target_owner_id), None)

    def sort_owners(self):
        self.all_my_owners = sorted(self.all_my_owners, key=lambda owner: owner.id)

File: Project Files/CarPark/Fish/Fish_JavaScript_Start/Extra/Aquarium.js
class Aquarium {
	constructor () {
		this.allMyFishOwners = []
	}

	addFishOwner  (newId, newFirstName, newLastName, newBirthDate) {
		var newFishOwner = new FishOwner(newId, newFirstName, newLastName, newBirthDate, this)
		this.allMyFishOwners.push(newFishOwner)
	}

	findFishOwner (targetOwnerId) {
		var foundOwner, aOwner
		foundOwner = null
		for (aOwner of this.allMyFishOwners) {
			if (aOwner.id === targetOwnerId) {
				foundOwner = aOwner
				break
			}
		}
		return foundOwner
	}

	sortFishOwners () {
		this.allMyFishOwners.sort(function (a, b) {
			return a.id > b.id
		})
	}

	getFishOwners () {
		var out, aFishOwner;
		this.sortFishOwners()
		out = ''
		for (aFishOwner of this.allMyFishOwners) {
			out += aFishOwner + '\n'
		}
		return out
	}

	getThoseWithTwoFishes  () {
		var out, aFishOwner;
		this.sortFishOwners()
		out = ''
		for (aFishOwner of this.allMyFishOwners) {
			if (aFishOwner.hasTwoFishes()) {
				out += aFishOwner + '\n'
				out += aFishOwner.getFishes()
			}
		}
		return out
	}
}